src/dmd/iasmdmd.d

```diff
@@ -4304,34 +4304,104 @@ JUMP_REF2:
             o1.ajt = ajt;
             break;
 
+        case TOK.bool_:
+            ptype = Type.tbool;
+            goto TYPE_REF;
+
+        case TOK.char_:
+            ptype = Type.tchar;
+            goto TYPE_REF;
+        case TOK.wchar_:
+            ptype = Type.twchar;
+            goto TYPE_REF;
+        case TOK.dchar_:
+            ptype = Type.tdchar;
+            goto TYPE_REF;
+
+        case TOK.uns8:
+            ptype = Type.tuns8;
+            goto TYPE_REF;
+        case TOK.uns16:
+            ptype = Type.tuns16;
+            goto TYPE_REF;
+        case TOK.uns32:
+            ptype = Type.tuns32;
+            goto TYPE_REF;
+        case TOK.uns64 :
+            ptype = Type.tuns64;
+            goto TYPE_REF;
+
         case TOK.int8:
             ptype = Type.tint8;
             goto TYPE_REF;
+        case ASMTKword:
+            ptype = Type.tint16;
+            goto TYPE_REF;
         case TOK.int32:
         case ASMTKdword:
             ptype = Type.tint32;
             goto TYPE_REF;
+        case TOK.int64:
+        case ASMTKqword:
+            ptype = Type.tint64;
+            goto TYPE_REF;
+
         case TOK.float32:
             ptype = Type.tfloat32;
             goto TYPE_REF;
-        case ASMTKqword:
         case TOK.float64:
             ptype = Type.tfloat64;
             goto TYPE_REF;
         case TOK.float80:
             ptype = Type.tfloat80;
             goto TYPE_REF;
-        case ASMTKword:
-            ptype = Type.tint16;
 TYPE_REF:
+
             bPtr = true;
             asm_token();
+            // <BasicType>.<min/max etc>
+            if (asmstate.tokValue == TOK.dot)
+                goto TYPE_PROP;
             asm_chktok(cast(TOK) ASMTKptr, "ptr expected");
             asm_cond_exp(o1);
             o1.ptype = ptype;
             o1.bPtr = bPtr;
             break;
 
+TYPE_PROP:
+            asm_token();
+            if (asmstate.tokValue == TOK.identifier)
+            {
+                TypeExp te = new TypeExp(asmstate.loc, ptype);
+                DotIdExp did = new DotIdExp(asmstate.loc, te, asmstate.tok.ident);
+                Scope *sc = asmstate.sc.startCTFE();
+                Expression e = did.expressionSemantic(sc);
+                sc.endCTFE();
+                e = e.ctfeInterpret();
+                if (e.isConst())
+                {
+                    if (e.type.isintegral())
+                    {
+                        o1.disp = e.toInteger();
+                    }
+                    else if (e.type.isreal())
+                    {
+                        o1.vreal = e.toReal();
+                        o1.ptype = e.type;
+                    }
+                    else
+                    {
+                        asmerr("bad type/size of operands `%s`", e.toChars());
+                    }
+                }
+            }
+            else
+            {
+                asmerr("basic type property expected");
```

Comment:
"type property" doesn't sound right.  I think "basic type expected" is what you mean.

Reply:
Finally done. Maybe you can give a second opinion because there 's still the icon showing that you request changes in the "Reviewers" frame at the top right corner.